File: Clases2023-24/Clase18-NavigationComponentFinal/app/src/main/java/com/example/clase18_navigationcomponentfinal/FragmentDos.java
package com.example.clase18_navigationcomponentfinal;

import static androidx.navigation.Navigation.findNavController;

import android.os.Bundle;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.example.clase18_navigationcomponentfinal.databinding.FragmentDosBinding;


public class FragmentDos extends Fragment {

    FragmentDosBinding binding;



    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        // Inflate the layout for this fragment
        binding= FragmentDosBinding.inflate(inflater,container,false);
        return binding.getRoot();
    }


    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);

        getParentFragmentManager().setFragmentResultListener("EnviarDato", getViewLifecycleOwner(), (requestKey, result) -> {
            binding.tvDatoRecibidoPorListener.setText("Dato recibido por listener: "+result.getString("dato"));
        });



        binding.tvDatoRecibido.setText("Dato recibido: "+getArguments().getString("dato"));

        binding.bFragDosIrAlTres.setOnClickListener(boton -> {
                findNavController(view).navigate(R.id.action_fragmentDos_to_fragmentTres);
        });

        binding.bFragDosIrAlUno.setOnClickListener(boton -> {
                findNavController(view).navigate(R.id.action_fragmentDos_to_fragmentUno);
        });


    }
}

File: Clases2024-25/Clase21_BottomNavigation_/app/src/main/java/com/example/clase21_bottomnavigation_/FragmentDos.kt
package com.example.clase21_bottomnavigation_

import android.os.Bundle
import androidx.fragment.app.Fragment
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import com.example.clase21_bottomnavigation_.databinding.FragmentDosBinding
import com.example.clase21_bottomnavigation_.databinding.FragmentUnoBinding


class FragmentDos : Fragment() {
    lateinit var binding: FragmentDosBinding

    override fun onCreateView(
        inflater: LayoutInflater, container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        binding= FragmentDosBinding.inflate(inflater, container, false)
        return binding.root
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)

        parentFragmentManager.setFragmentResultListener("dato", this) { requestKey, bundle ->
            val datoRecibido = bundle.getString("dato")
            binding.tvDatoRecibido.text = datoRecibido ?: "No se recibió ningún dato"
        }




    }
}